docs/config.md

```diff
@@ -0,0 +1,159 @@
+<!--
+Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Model Analyzer Config
+
+In addition to the CLI flags, you can config the Model Analyzer
+using a [YAML](https://yaml.org/) file too. The scheme
+for this configuration file is described below. Brackets
+indicate that a parameter is optional. For non-list
+and non-object parameters the value is set to the
+specified default.
+
+The placeholders below are used throughout the configuration:
+
+* `<boolean>`: a boolean that can take `true` or `false` as value.
+* `<string>`: a regular string
+* `<comma-delimited-list>`: a list of comma separated items.
+* `<int>`: a regular integer value.
+* `<list>`: a list of values.
+* `<range>`: An object containing `start` and `end` key with an optional `step` value. The types that support `<range>` can be described by a list or using the example structure below:
+```yaml
+batch_sizes:
+    start: 2
+    end: 6
+    step: 2
+```
+
+This YAML represents the array `[2, 4]`. Note that `end` is not included.
```

Comment:
Somehow, this doesn't make sense to me. If I have a start of 2, end of 6, and step of 2, I would expect this to run [2,4,6]. Why is `end` not included?

Reply:
Also, have you considered specifying the range like... `[start:step:stop]`. For example a value of `[1:1:6]` would create an array of `[1,2,3,4,5,6]`, or `[3:5:15]` will generate `[3,8,13]`

Reply:
I will fix the config parsing to include the last value too. Regarding the alternative format, it would make it a bit difficult because of the way we parse configs (we parse configs according to the different types that they have) but I can rename  `end` to `stop` so that it is more intuitive.

Reply:
Just a thing to think about in the future.